### ISSUE TITLE
SendPingAsyncWithHostAndTimeoutAndBufferAndPingOptions failed on Win81

### ISSUE THREAD

**Issue by danmosemsft** [dotnet/corefx#35591]
SendPingAsyncWithHostAndTimeoutAndBufferAndPingOptions and SendPingAsyncWithHostAndTimeoutAndBuffer

@wfurt is there more information we can put in the exception beyond `An exception occurred during a Ping request.` such as eg an error code or more detailed text?
```
Windows.81.Amd64.Open-x86-Release
Get Repro environment
Unhandled Exception of Type System.Net.NetworkInformation.PingException
Message :
System.Net.NetworkInformation.PingException : An exception occurred during a Ping request.
---- System.ComponentModel.Win32Exception : The operation completed successfully.
Stack Trace :
   at System.Net.NetworkInformation.Ping.GetAddressAndSendAsync(String hostNameOrAddress, Int32 timeout, Byte[] buffer, PingOptions options) in D:\a\1\s\src\System.Net.Ping\src\System\Net\NetworkInformation\Ping.cs:line 408
   at System.Net.NetworkInformation.Tests.PingTest.SendPingAsync(Func`2 sendPing, Action`1 pingResultValidator) in D:\a\1\s\src\System.Net.Ping\tests\FunctionalTests\PingTest.cs:line 750
   at System.Net.NetworkInformation.Tests.PingTest.SendPingAsyncWithHostAndTimeoutAndBuffer() in D:\a\1\s\src\System.Net.Ping\tests\FunctionalTests\PingTest.cs:line 458
--- End of stack trace from previous location where exception was thrown ---
----- Inner Stack Trace -----
   at System.Net.NetworkInformation.Ping.DoSendPingCore(IPAddress address, Byte[] buffer, Int32 timeout, PingOptions options, Boolean isAsync) in D:\a\1\s\src\System.Net.Ping\src\System\Net\NetworkInformation\Ping.Windows.cs:line 92
   at System.Net.NetworkInformation.Ping.SendPingAsyncCore(IPAddress address, Byte[] buffer, Int32 timeout, PingOptions options) in D:\a\1\s\src\System.Net.Ping\src\System\Net\NetworkInformation\Ping.Windows.cs:line 42
   at System.Net.NetworkInformation.Ping.GetAddressAndSendAsync(String hostNameOrAddress, Int32 timeout, Byte[] buffer, PingOptions options) in D:\a\1\s\src\System.Net.Ping\src\System\Net\NetworkInformation\Ping.cs:line 395
```
https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F35588~2Fmerge/test~2Ffunctional~2Fcli~2F/20190226.6/workItem/System.Net.Ping.Functional.Tests

**Comment by danmosemsft** [comment]
This is also bad -- `The operation completed successfully.` -- we should ideally somehow not throw a Win32 exception if there is no error code..

**Comment by wfurt** [comment]
This looks like dupe of  #35351 
I start looking at it and it seems like the GetLastError() really _does_ return 0. I'll keep looking. It seems like something changed recently and it is not the ping code.

**Comment by wfurt** [comment]
probably also related to https://github.com/dotnet/coreclr/pull/22871

**Comment by wfurt** [comment]
fixed by #33621 and https://github.com/dotnet/coreclr/pull/22871
